code/obj/item/device/nuclear_op_items.dm

```diff
@@ -0,0 +1,824 @@
+
+	airstrike(atom/target, params, mob/user, reach)
+		..()
+		src.desc = "A handheld monocular device with a laser built into it, used for calling in fire support from the Cairngorm. It has [src.uses] charge left."
+		return TRUE
+
+	pixelaction(atom/target, params, mob/user, reach, continuousFire = 0)
+		if (reach)
+			return FALSE
+		if (!isturf(user.loc))
+			return FALSE
+		if (uses <= 0)
+			return FALSE
+		if (in_use)
+			return FALSE
+		if(target.z != 1 || user.z != 1)
+			return
+
+		for_by_tcl(A, /obj/machinery/broadside_gun)
+			var/obj/machinery/broadside_gun/C = A
+			if(C.firingfrom == src.ship_looking_for && !C.broken && ((C.ammo <= 0) || (!C.ammo == -1)))
+				src.linked_gun = C
+				break
+
+		if(!src.linked_gun)
+			boutput(user, "<span class='alert'>The [src] makes a grumpy beep. It seems there's no artillery guns in position currently.</span>")
+			playsound(src, "sound/machines/buzz-sigh.ogg", 50, 1)
+			return FALSE
+
+		return src.airstrike(target, params, user, reach)
+
+
+
+/obj/machinery/broadside_gun //Thanks to Cogwerks for the sprites
+	name = "Broadside Gun Parent"
+	icon = 'icons/obj/large/96x32.dmi'
+	icon_state = "artillery_cannon"
+	desc = "Parent of broadside guns for fire support."
+	density = 1
+	anchored = 1
+	processing_tier = PROCESSING_EIGHTH
+	bound_width = 96
+	/// Ship name you're firing from, important for the designator
+	var/firingfrom = ""
+	var/broken = FALSE
+	/// Amount of ammo the gun has, set to -1 for infinite
+	var/ammo = 1
+	/// In case you need to offset the gun firing's sound by offset tiles (if it's aiming left for example)
+	var/sound_offset_length
+	/// In case you need to offset the gun firing's sound dir (if it's aiming left for example)
+	var/sound_offset_dir
+	/// Holding var for the exact turf to play the gun's firing sound from
+	var/turf/sound_turf
+	/// Overlay sprite for where the strike will land, set to null for no overlay
+	var/image/target_overlay = null
+
+
+	proc/bombard(var/atom/target, var/mob/user)
+		return
+
+	New()
+		. = ..()
+		START_TRACKING
+		target_overlay = image('icons/effects/effects.dmi', "spinny_red")
+		sound_turf = get_turf(src)
+
+	disposing()
+		. = ..()
+		STOP_TRACKING
+		target_overlay = null
+
+/obj/machinery/broadside_gun/artillery_cannon
+	name = "Artillery Cannon"
+	icon = 'icons/obj/large/96x32.dmi'
+	icon_state = "152mm"
+	desc = "A 152 millimeter artillery cannon, used for heavy fire support."
+	bound_width = 96
+	firingfrom = ""
+	ammo = -1
+	sound_offset_dir = EAST
+	sound_offset_length = 3
+
+	bombard(atom/target, mob/user)
+		var/turf/target_turf = get_turf(target)
+		var/turf/firing_turf = get_turf(src)
+		if(getlineopaqueblocked(get_turf(user), target_turf) != target_turf)
+			return FALSE
+		if(!isnull(src.target_overlay))
+			target_turf.overlays += src.target_overlay
+		while(sound_offset_length > 0)
+			sound_turf = get_step(src, sound_offset_dir)
+			sound_offset_length--
```

Comment:
this looks weirdly useless

Reply:
The gun faces left, and so the sound will always be on the end of the barrel, which is in space, which means people nearby can't hear it.